presto-main/src/test/java/com/facebook/presto/execution/buffer/TestBroadcastOutputBuffer.java

```diff
@@ -943,6 +944,27 @@ public void testBufferCompletion()
         assertTrue(buffer.isFinished());
     }
 
+    @Test
+    public void testBufferFinishesWhenClientBuffersDestroyed()
+    {
+        BroadcastOutputBuffer buffer = createBroadcastBuffer(
+                createInitialEmptyOutputBuffers(BROADCAST)
+                        .withBuffer(FIRST, BROADCAST_PARTITION_ID)
+                        .withBuffer(SECOND, BROADCAST_PARTITION_ID)
+                        .withBuffer(THIRD, BROADCAST_PARTITION_ID)
+                        .withNoMoreBufferIds(),
+                sizeOfPages(5));
+
+        // the buffer is in the NO_MORE_BUFFERS state now
+        // and if we abort all the buffers it should destroy itself
+        // and move to the FINISHED state
+        buffer.abort(FIRST);
```

Comment:
Add some pages before closing everything.  This ensures that the buffer closes even if there are pending pages.